crates/bifrost/src/read_stream.rs

```diff
@@ -102,11 +116,45 @@ enum State {
         #[pin]
         tail_watch: Option<BoxStream<'static, TailState>>,
     },
-    /// Waiting for the tail LSN of the substream's loglet to be determined (sealing in-progress)
-    AwaitingReconfiguration,
+    /// Chain reconfiguration has been detected, we'll update our view of the chain.
+    Reconfiguring,
+    /// Waiting for the tail LSN of the substream's loglet to be determined (sealing in-progress).
+    /// We land on this state when we observe a "Sealed" signal from an open segment after we
+    /// reached the safe known tail for that loglet. We can continue reading only after the chain
+    /// has been sealed by a marker or with a new installed chain.
+    AwaitingReconfiguration {
```

Comment:
Nit: `Reconfiguring` (compared to `AwaitingReconfiguration`) reads as if the read stream would reconfigure something. Could renaming `Reconfiguring` into` Awaiting(NewSegment | Reconfiguration)` and `AwaitingReconfiguration` into `Awaiting(NewSegment | Reconfiguration)OrSealChain` be an idea?

Reply:
No strong opinions here. I'll update.